pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -464,6 +466,20 @@ func translateRoute(
 	if in.CorsPolicy != nil {
 		out.TypedPerFilterConfig[wellknown.CORS] = protoconv.MessageToAny(TranslateCORSPolicy(in.CorsPolicy))
 	}
+	if hostname.IsSet() {
+		statefulSvc := serviceRegistry[hostname]
+		if statefulConfig := util.MaybeBuildStatefulSessionFilterConfig(statefulSvc); statefulConfig != nil {
```

Comment:
so now we only build it when at least a vs is there?

Reply:
Yes. That is the change so that we can predictably find the service based on destination. This is to align with the model of moving this to API in future.

Reply:
cannot we use the stateful session feature with no VS set now, while we can previously

Reply:
Yes. That is the change. As I said, it is to align with future API we may introduce.

Reply:
Is it intended that stateful session must have a VS associated, I am not very convinced. I can understand that we may need a session sticky cap per route for customized matching. 
 @costinm Please take a look at this

Reply:
It there is no VS associated, how would customer specify the stickyness per route without VS?

Reply:
Yes, I think we need both

Reply:
Do you have any idea of how would users use it when we move this to API? IMO this is similar to consistent hashing API for which VS is needed.

Reply:
@costinm WDYT?

Reply:
@howardjohn @hzxuzhonghu @costinm WDYT? I am OK to put it back to non virtual service services but want to keep the long term API in mind so that these labels can be moved easily to API (I am thinking it should be some what similar to consistent hashing API)

Reply:
In terms of future API - I think 'statefullness' will need to be a policy attached to backend services ( destinations ). The Gateway WG already has an API defined - I don't think we should have a duplicate one for Istio, but we can keep the label for compat and until the K8S gateway makes its way to clusters and to beta. 

The main requirement is that users should be able to express that a Service ( == destination ) requires the special cookie   handling.

Reply:
I also don't think we can or should implement the 'attach to route' part of the gateway API - we can wait until others implement it and see how it works in real life, since the semantics are very confusing and under-documented. I am mainly concerned of using the API as a Service (backendRef) policy.